### PR TITLE
feat: Add extensive unit tests for core, mechanics, game_state, and U…

### PR DESCRIPTION
**Pull request by woozyrabbit123**
…I actions

This commit culminates a comprehensive review and test expansion phase, significantly increasing unit test coverage across critical game modules. This effort has improved code robustness, identified and fixed several bugs, and provides a stronger foundation for future development.

**Key Areas Enhanced:**

1.  **`src/core` Modules:**
    *   **`drug.py`**: Added tests for `Drug` class initialization and `get_quality_multiplier`. Fixed a bug in multiplier logic for unknown price types.
    *   **`player_inventory.py`**: Extended tests beyond skill unlocking to cover drug management (add/remove, capacity, edge cases), crypto management (add/remove, edge cases), capacity calculations, initialization, and `formatted_summary`.
    *   **`region.py`**: Added extensive tests for initialization, heat effects, market setup, complex price/stock calculations (including various event impacts via mocking), stock updates, and restocking.
    *   **`ai_rival.py`**: Added tests for `AIRival` data class initialization and 'busted' state.
    *   **`market_event.py`**: Added tests for `MarketEvent` initialization across various event types and for the `__str__` method.

2.  **`src/mechanics` Modules:**
    *   **`market_impact.py`**:
        *   **Fixed/Implemented**: Ensured `apply_player_sell_impact` correctly applies the `COMPARTMENTALIZATION_HEAT_REDUCTION_PERCENT` from `game_configs` to reduce market impact factor when the skill is active. Updated function signature and call site in `ui_pygame/actions.py`.
        *   Added tests for `apply_player_buy_impact`, `apply_player_sell_impact` (with/without Compartmentalization), and `decay_player_market_impact`.
    *   **`event_manager.py`**:
        *   Extended tests for event creation functions (`_create_and_add_...` for Demand Spike, Supply Disruption, Police Crackdown, Cheap Stash, The Setup, Rival Busted, Drug Market Crash, Black Market). Ensured correct `MarketEvent` instantiation and attribute assignment using mocks for randomness. (Mugging and Forced Fire Sale event handlers were tested previously).

3.  **`src/game_state.py`:**
    *   Added tests for `initialize_game_state`, `update_daily_crypto_prices` (including volatility, min price logic, edge cases), and `initialize_global_state` (testing both sub-initializer calls and integrated effects).

4.  **`src/ui_pygame/actions.py`:**
    *   Significantly expanded tests for:
        *   `action_confirm_transaction` (buy part): Added success and failure cases (no cash, no stock, no capacity).
        *   `action_confirm_tech_operation`: Added numerous tests covering various operations (buy/sell crypto, launder cash, stake/unstake DrugCoin) for success and common failure scenarios (invalid input, insufficient funds/items, wrong item type). Heat calculation with OpSec skills for crypto operations was also implicitly covered via these tests.
    *   Refactored Pygame UI state mocking in tests for better consistency.

**Overall Impact:**

*   The total number of unit tests increased from a small initial set to 106.
*   All 106 tests are passing.
*   Several bugs and areas for improvement were identified and addressed during this process, leading to more reliable code.

### CODE REVIEW

File: src/mechanics/market_impact.py [hunk 1]
```diff
@@ -14,51 +16,23 @@ def apply_player_buy_impact(region: Region, drug_name_enum: DrugName, quantity_b
     new_modifier = min(current_modifier + impact_factor, 1.25) 
     region.drug_market_data[drug_name_enum]["player_buy_impact_modifier"] = new_modifier
 
-def apply_player_sell_impact(player_inv: PlayerInventory, region: Region, drug_name_enum: DrugName, quantity_sold: int):
+def apply_player_sell_impact(
```
[the remark this answers is not in the document]
Comment (woozyrabbit123) [reply src/mechanics/market_impact.py:19]: There is a substantial block of commented-out code regarding the skill check logic. Consider removing or summarizing these comments to improve code readability and maintainability.
